src/EA.Weee.RequestHandlers.Tests.DataAccess/Shared/CommonDataAccessTests.cs

```diff
@@ -75,5 +76,42 @@ public async Task GetCompetentAuthorityById_AuthorityShouldBeReturned(Guid id, s
                 result.Abbreviation.Should().Be(abbreviation);
             }
         }
+
+        [Theory]
+        [InlineData(Core.Shared.CompetentAuthority.England, "EA")]
+        [InlineData(Core.Shared.CompetentAuthority.Wales, "NRW")]
+        [InlineData(Core.Shared.CompetentAuthority.Scotland, "SEPA")]
+        [InlineData(Core.Shared.CompetentAuthority.NorthernIreland, "NIEA")]
+        public async Task FetchCompetentAuthorityApprovedSchemes_AuthorityShouldBeReturned(Core.Shared.CompetentAuthority authority, string abbreviation)
+        {
+            using (var database = new DatabaseWrapper())
+            {
+                var helper = new ModelHelper(database.Model);
+
+                var dataAccess = new CommonDataAccess(database.WeeeContext);
+
+                var result = await dataAccess.FetchCompetentAuthorityApprovedSchemes(authority);
+
+                result.Abbreviation.Should().Be(abbreviation);
+            }
+        }
+
+        [Fact]
+        public async Task FetchCompetentAuthorityApprovedSchemes_OnlyApprovedSchemesReturned()
+        {
+            using (var database = new DatabaseWrapper())
+            {
+                var helper = new ModelHelper(database.Model);
+                var authorityId = Guid.Parse("A3C2D0DD-53A1-4F6A-99D0-1CCFC87611A8");
+
+                var dataAccess = new CommonDataAccess(database.WeeeContext);
+
+                await database.Model.SaveChangesAsync();
+
+                var result = await dataAccess.FetchCompetentAuthorityApprovedSchemes(Core.Shared.CompetentAuthority.England);
+
+                result.Schemes.Should().NotContain(x => x.SchemeStatus.Value == SchemeStatus.Rejected.Value);
```

Comment:
would you mind adding a test where  result.Schemes.Should().Contain(status.Value == Approved)

Reply:
Added